market/models.py

```diff
@@ -170,3 +195,31 @@ class Meta:
 
     def __str__(self):
         return f"{self.market.market_id}[{self.round}]"
+
+class UnusedCosts(models.Model):
+    market = models.ForeignKey(Market, on_delete=models.CASCADE)
+    cost = models.DecimalField(max_digits=14, decimal_places=2, 
+        validators=[MinValueValidator(Decimal('0.01'))])
+
+    class Meta:
+        # There cannot be unused costs of the same value in the same market
+        constraints = [
+            models.UniqueConstraint(fields=['market', 'cost'], name='market_and_unused_cost_unique_together'),
```

Comment:
If max_cost = min_cost,  all costs are identical. So this constraint is problematic. Same goes for the constraint on UsedCosts below. 

Either the constraints should be removed, or (perhaps better) the new algorithm and the two new tables should only be used when max_cost > min_cost.